pkg/api/register.go

```diff
@@ -111,6 +111,15 @@ func AddToScheme(scheme *runtime.Scheme) {
 		&unversioned.APIGroup{},
 		&unversioned.APIResourceList{},
 	)
+
+	// Types registered here will be kept unversioned when they are encoded.
+	// It's for backward compatibility with release 1.1. clients.
+	Scheme.AddToKeepUnverioned(
```

Comment:
So client impl is easier because we can extend this method to take a group/version and record that alongside the decision to remove the flag.  Server impl needs an explicit wrapper to fix discovery.